python/cudf/dataframe/index.py

```diff
@@ -455,7 +455,6 @@ def __init__(self, values, name=None):
             values = NumericalColumn(data=Buffer(values), dtype=values.dtype)
```

Comment:
If we're removing the assertion we need to update this function to construct the `NumericalColumn` in a way that calculates the nullmask as opposed to just creating the Buffer directly here.
```suggestion
            values = columnops.as_column(values)
            assert isinstance(values, NumericalColumn)
```